### PR TITLE
Use installed packages for impact analysis

### PR DESCRIPTION
**Pull request by simao**
https://advancedtelematic.atlassian.net/browse/PRO-1423

### CODE REVIEW

File: core/src/test/scala/org/genivi/sota/core/ImpactResourceSpec.scala [hunk 1]
```diff
@@ -39,11 +45,15 @@ class ImpactResourceSpec
...
+
+    val route = new ImpactResource(defaultNamespaceExtractor, fakeExternalResolver(affected)).route
+
     Get("/impact/blacklist") ~> route ~> check {
       status shouldBe StatusCodes.OK
-      val resp = responseAs[Seq[Map[Device.Id, PackageId]]]
+      val resp = responseAs[Map[Device.Id, Seq[PackageId]]]
 
-      resp should contain(Map(device.id → pkg.id))
+      resp should contain(device.id -> Seq(pkg.id))
```
Comment (CalumMcCall) [inline core/src/test/scala/org/genivi/sota/core/ImpactResourceSpec.scala]: Is `shouldEqual` not better here, so we ensure that only affected devices are actually returned?

File: external-resolver/src/main/scala/org/genivi/sota/resolver/db/PackageRepository.scala [hunk 2]
```diff
@@ -2,19 +2,27 @@
...
 import org.genivi.sota.refined.SlickRefined._
 import org.genivi.sota.resolver.common.Errors
-import org.genivi.sota.resolver.filters.{Filter, FilterRepository}
+import slick.driver.MySQLDriver.api._
 
 import scala.concurrent.ExecutionContext
-import slick.driver.MySQLDriver.api._
 
 
+object Wat {
```
Comment (CalumMcCall) [inline external-resolver/src/main/scala/org/genivi/sota/resolver/db/PackageRepository.scala]: Wat?

File: external-resolver/src/test/scala/org/genivi/sota/resolver/test/PackagesResourceSpec.scala [hunk 3]
```diff
@@ -74,6 +75,39 @@ class PackagesResourcePropSpec extends ResourcePropSpec with PackageGenerators {
...
+        status shouldBe StatusCodes.OK
+      }
+
+      installPackage(device, p) ~> route ~> check {
+        status shouldBe StatusCodes.OK
+      }
+
+      getAffected(Set(p.id)) ~> route ~> check {
+        status shouldBe StatusCodes.OK
+        responseAs[Map[Device.Id, Seq[PackageId]]] should contain(device -> Seq(p.id))
```
Comment (CalumMcCall) [inline external-resolver/src/test/scala/org/genivi/sota/resolver/test/PackagesResourceSpec.scala]: As above, I think matching exactly might be better here, to minimize the chance of false test passes.

**Comment by simao** [comment]
addressed those comments.

**Pull request merged**